types/tx.go

```diff
@@ -16,15 +17,14 @@ import (
 // Might we want types here ?
 type Tx []byte
 
+// Key produces a fixed-length key for use in indexing.
+func (tx Tx) Key() TxKey { return sha256.Sum256(tx) }
+
 // Hash computes the TMHASH hash of the wire encoded transaction.
-func (tx Tx) Hash() []byte {
-	return tmhash.Sum(tx)
-}
+func (tx Tx) Hash() []byte { return tmhash.Sum(tx) }
```

Comment:
Not something we have to change now, but I think it'd be preferable to add these as private fields on the `Tx` struct that we set the first time we run this `Hash` method and then check on subsequent calls to the `Hash` function to cut down on time spent recomputing the same data.

Reply:
I agree.

Reply:
Note that internally we wrap tx's in the `WrappedTx` struct which caches the hash among other things.